src/Datadog.Trace/Logging/CustomSerilogLogProvider.cs

```diff
@@ -27,15 +28,20 @@ public IDisposable OpenContext(object enricher)
 
         internal static Type GetLogEnricherType() => Type.GetType("Serilog.Core.ILogEventEnricher, Serilog");
 
-        private static Func<object, IDisposable> GetPush()
+        internal static new bool IsLoggerAvailable() =>
+            SerilogLogProvider.IsLoggerAvailable() && GetPushMethodInfo() != null;
+
+        private static MethodInfo GetPushMethodInfo()
         {
             var ndcContextType = Type.GetType("Serilog.Context.LogContext, Serilog");
+            return ndcContextType?.GetMethod("Push", GetLogEnricherType());
+        }
 
-            var logEventEnricherType = GetLogEnricherType();
-
-            var pushPropertyMethod = ndcContextType.GetMethod("Push", logEventEnricherType);
+        private static Func<object, IDisposable> GetPush()
+        {
+            var pushPropertyMethod = GetPushMethodInfo();
```

Comment:
Should we add more error-checking in case any of these reflection calls fail? e.g. if `pushPropertyMethod` is `null`?

Reply:
We could if we want to be extra careful...but I don't think it's needed. If `GetPushMethodInfo` returns `null`, it should never have gotten to this point because `IsLoggerAvailable()`, which is the first check before instantiating and using this logger, would have returned `false`